### PR TITLE
Initial Fastly CICD pipeilne and Services:Domain resource

### PR DESCRIPTION
**Pull request by mmaeng**
Initial CICD pipeline for fastly and the build for Fastly:Services:Domain

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/cicd.yml [hunk 1]
```diff
@@ -191,7 +190,6 @@ Resources:
               - iam:DeleteRolePolicy
               - iam:GetRolePolicy
               - iam:PassRole
-              - iam:UpdateAssumeRolePolicyDocument
```
Comment (ericzbeard) [inline release/cicd.yml:194]: Why was this removed? Pretty sure this will break something.

File: release/fastly/cicd.yml [hunk 2]
```diff
@@ -0,0 +1,735 @@
...
+        Image: !Sub "${AWS::AccountId}.dkr.ecr.${AWS::Region}.amazonaws.com/cep-cicd:latest"
+        ImagePullCredentialsType: SERVICE_ROLE
+        PrivilegedMode: true
+        Type: LINUX_CONTAINER
+        EnvironmentVariables:
+        - Name: RESOURCE_PATH
+          Type: PLAINTEXT
+          Value: "placeholder-for-path-to-resource"
+        - Name: FASTLY_SERVICE_ID
+          Type: SECRETS_MANAGER
```
Comment (ericzbeard) [inline release/fastly/cicd.yml:204]: We should consolidate on parameter store for values that are not sensitive. I'm using secrets manager for typeConfiguration, which has actual secrets in it.

File: release/fastly/cicd.yml [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+            Principal:
+              AWS: 
+                - !Sub "arn:aws:iam::${BetaAccountId}:root"
+                - !GetAtt PublishCrossAccountRole.Arn
+            Action: s3:PutObject
+            Resource: 
+              - !Sub "arn:aws:s3:::cep-source-${ProdAccountId}-prod-${PrefixLower}"
+              - !Sub "arn:aws:s3:::cep-source-${ProdAccountId}-prod-${PrefixLower}/*"
+
+  StackSetAdministrationRole:
```
Comment (ericzbeard) [inline release/fastly/cicd.yml:564]: Should be safe to delete the StackSet resources, since we aren't publishing with stack sets any more.

**Review by ericzbeard — CHANGES_REQUESTED** [review]

File: release/cicd.yml (hunk 1 above)
Comment (mmaeng) [reply release/cicd.yml:194]: The cloudformation console is saying iam:UpdateAssumeRolePolicyDocument doesn't exist as a property so I removed it.

Comment (ericzbeard) [reply release/cicd.yml:194]: Weird, I only added those when I got errors for actions that were missing. I guess it should be `UpdateAssumeRolePolicy`

File: release/fastly/cicd.yml (hunk 2 above)
Comment (mmaeng) [reply release/fastly/cicd.yml:204]: My main motivation was the built in regional replication for secrets manager.  One less item to maintain but I can switch it.

Comment (ericzbeard) [reply release/fastly/cicd.yml]: We don't need cross region, since we aren't deploying a pipeline per region any more.

File: release/cicd.yml [hunk 4]
```diff
@@ -164,7 +164,6 @@ Resources:
             Resource: "*"
           - Action:
               - kms:CreateKey
-              - kms:DeleteKey
```
Comment (ericzbeard) [inline release/cicd.yml:167]: Why was this removed?

**Review by ericzbeard — CHANGES_REQUESTED** [review]
I just merged the common template and now there's a conflict in the cicd template.

File: release/cicd.yml (hunk 4 above)
Comment (mmaeng) [reply release/cicd.yml:167]: Deletekey also doesnt look to be an existing property, it looks to have been replaced with scheduledeletion.

Comment (mmaeng) [reply release/cicd.yml:167]: Let me revert this change for this PR.

**Comment by mmaeng** [comment]
Included in #116

**Pull request closed without merging**